pkg/api/vlabs/validate.go

```diff
@@ -1122,6 +1122,28 @@ func (k *KubernetesConfig) Validate(k8sVersion string, hasWindows bool) error {
 	if e := k.validateNetworkPluginPlusPolicy(); e != nil {
 		return e
 	}
+	if e := k.validatePrivateAzureRegistryServer(); e != nil {
+		return e
+	}
+
+	return nil
+}
+
+func (k *KubernetesConfig) validatePrivateAzureRegistryServer() error {
+
+	// Check PrivateAzureRegistryServer has a valid value.
+	valid := false
+	if k.PrivateAzureRegistryServer != "" {
+		if k.CustomHyperkubeImage != "" {
+			valid = true
+		}
+	} else {
+		valid = true
+	}
```

Comment:
I found this logic hard to read.  would something like the following be more more readable?  I think that would remove the need for the `valid` check

```go
k.PrivateAzureRegistryServer != "" and k.CustomHyperkubeImage == "" {
    return errors.Errorf("customHyperkubeImage must be provided when privateAzureRegistryServer is provided")
}
```